sidecar/reloader/main.go

```diff
@@ -160,31 +120,64 @@ func main() {
 		}
 	}()
 
-	// plugin config
-	for _, file := range pluginConfigList {
-		fullFile := filepath.Join(pluginConfigDir, file)
-		if _, err := os.Stat(fullFile); err == nil {
-			log.Printf("add file %s to watcher\n", fullFile)
-			err = watcher.Add(fullFile)
-			if err != nil {
-				log.Fatal(err)
-			}
-			fileCheck[fullFile] = getMD5(fullFile)
+	// generate watched file list
+	fileList := generateWatchedFileList(pluginConfigDir, licenseFilePath)
+
+	// add file to watcher and return fileCheck result
+	fileCheck = addAllFileToWatcher(watcher, fileList)
+
+	done := make(chan bool)
+	<-done
+}
+
+func generateWatchedFileList(pluginConfigDir, licenseFilePath string) []string {
```

Comment:
Can we running `os.GetEnv` in this function?, because there are not other function use `pluginConfigDir` and `licenseFilePath `

Reply:
licenseFilePath  has used in select code block

Reply:
Sure